File: skypydb/api/mixins/vector/list_collections.py
"""Mixin that lists vector collections for `VectorClient`."""

from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
    from skypydb.api.collection import Collection


class ListCollectionsMixin:
    """Expose collection listing behavior on `VectorClient`."""

    def list_collections(self) -> List["Collection"]:
        """Return all available collections as cached wrappers."""

        collections = []
        for collection_info in self._db.list_collections():
            name = collection_info["name"]
            if name in self._collections:
                collections.append(self._collections[name])
                continue

            from skypydb.api.collection import Collection

            collection = Collection(
                db=self._db,
                name=name,
                metadata=collection_info.get("metadata"),
            )
            self._collections[name] = collection
            collections.append(collection)
        return collections

File: skypydb-js/src/api/mixins/vector/list_collections.ts
import type { CollectionInfo } from "../../../types";
import { Collection } from "../../collection";
import { GetCollectionMixin } from "./get_collection";
import type { VectorDatabase } from "../../../database/vector_db";

export class ListCollectionsMixin extends GetCollectionMixin {
  async list_collections(): Promise<Collection[]> {
    const collections: Collection[] = [];
    for (const collection_info of this._db.list_collections()) {
      const cached = this._collections.get(collection_info.name);
      if (cached) {
        collections.push(cached);
        continue;
      }
      const collection = new Collection(
        this._db,
        collection_info.name,
        collection_info.metadata,
      );
      this._collections.set(collection_info.name, collection);
      collections.push(collection);
    }
    return collections;
  }
}
